torch/_inductor/fx_passes/split_cat.py

```diff
@@ -48,20 +57,125 @@ def apply(self, match, graph, node):
         if any(isinstance(section, torch.SymInt) for section in split_sections):
             # TODO dynamic_shapes with assume_static_by_default=False fails while AOT Autograd tracing.
             return
+        if split_dim < 0:  # Normalize split dim
+            split_dim += split_input.meta["example_value"].dim()
         with graph.inserting_after(split_node):
             new_split_node = graph.call_function(
-                torch.split, args=(split_input, split_sections, split_dim)
+                torch.split,
+                args=(split_input, split_sections),
+                kwargs={"dim": split_dim},
             )
         split_node.replace_all_uses_with(new_split_node)
         new_split_node.meta.update(split_node.meta)
         graph.erase_node(split_node)
         counters["inductor"]["split_cat_norm"] += 1
 
 
+class TorchSplit(CallFunction):
+    """
+    Matches a call to torch.split if it is in a normalized form. Ensures that all users of
+    splits are unique getitems.
+    """
+
+    def __init__(self, arg):
+        # using KeywordArg("dim") for `dim` checks they all match
+        super().__init__(
+            torch.split, arg, Arg(), _users=MULTIPLE, dim=KeywordArg("dim")
+        )
+
+    def _match(self, node: torch.fx.Node, ctx: MatchContext):
+        m = super()._match(node, ctx)
+        if not m:
+            return m
+        split_sections = get_arg_value(node, 1, "split_size_or_sections")
+        if not isinstance(split_sections, (list, tuple)):
+            return FailedMatch("split not normalized")
+        # check users are all unique getitems
+        seen_idxs = set()
+        for user in node.users:
+            if not match_node(user, {operator.getitem}):
+                # This should ideally never happen. Split user should always be a getitem
+                return FailedMatch(f"user of split not a getitem: {user}")
+            if user.args[1] in seen_idxs:
```

Comment:
Should also check this is an integer.

Reply:
What should I do if it is a non-int? I was assuming the 2nd input to getitem will always be an int. Since I don't handle non-ints, should I just return a FailedMatch for symint like cases?

Reply:
Just return `FailedMatch`